File: client/src/main/java/in/clayfish/printful/models/Response.java
package in.clayfish.printful.models;

import java.io.Serializable;
import java.util.List;

import in.clayfish.printful.models.includable.Paging;

/**
 * @author shuklaalok7
 * @since 21/12/2016
 */
public class Response<T> implements Serializable {
    /**
     * Response status code
     */
    private int code;

    /**
     *
     */
    private String errorMessage;

    /**
     * List of data
     */
    private List<T> result;

    /**
     * Paging information
     */
    private Paging paging;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public List<T> getResult() {
        return result;
    }

    public void setResult(List<T> result) {
        this.result = result;
    }

    public Paging getPaging() {
        return paging;
    }

    public void setPaging(Paging paging) {
        this.paging = paging;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}

File: client/src/main/java/in/clayfish/printful/models/Response.kt
/*
 * MIT License
 *
 * Copyright (c) 2016-2019 ClayFish Technologies LLP
 * Copyright (c) 2020 ClayFish Technologies
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package `in`.clayfish.printful.models

import `in`.clayfish.printful.models.includable.Paging
import java.io.Serializable

/**
 * @author shuklaalok7
 * @since 21/12/2016
 */
data class Response<T>(
        /**
         * Response status code
         */
        var code: Int = 0,

        /**
         *
         */
        var errorMessage: String? = null,

        /**
         * List of data
         */
        var result: List<T>? = null,

        /**
         * Paging information
         */
        var paging: Paging? = null
) : Serializable
